### ISSUE TITLE
Game freezes when window is closed on Mac on LibGDX 1.11.0 and LWJGL3

### ISSUE THREAD

**Issue by rotter** [libgdx/libgdx#6872]
#### Issue details
After I launch the game on Android Studio (DesktopLauncher.main) if I close the game using Cmd-Q, I get randomly one of these outcomes:
1) freezes for a few seconds and then process stops gracefully; or
2) freezes for several minutes (maybe/probably indefinitely, didnt wait much to test, I force STOP after a while); or
3) locks Android Studio completely (happened only once, might be unrelated, but it feels like it is related)

#### Reproduction steps/code
This issue is happening only after migrating to LibGDX 1.11.0 and LWJGL3, and never happened before (ie, on 1.10 & non-LWJGL3). This happens on the default generated project (badlogic.jpg one), the only change I have to do is add the extension to gradle: gdx-lwjgl3-glfw-awt-macos . I made no other change to the project.

#### Version of libGDX and/or relevant dependencies
Latest OSX. Non-M1 mac. LibGDX 1.11.0 with gdx-lwjgl3-glfw-awt-macos extension.

#### Stacktrace
EventQueue class ->  getNextEvent() method -> freezes on pushPopCond.await()

Threads:
"AWT-EventQueue-0"@1,047 in group "main": WAIT
"AppKit Thread"@989: RUNNING
"AWT-Shutdown" @992: WAIT
"Common-Cleaner" @419 in group "InnocuousThreadGroup": WAIT
"DestroyJavaVM"@1,681 in group "main": RUNNING
"Finalizer" @1,683: WAIT
"Reference Handler" @1,682: RUNNING
"Signal Dispatcher" @1,684: RUNNING

#### Please select the affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [X] macOS

**Comment by kyu-n** [comment]
Does this issue occur with LWJGL3's own glfw_async library? You can load it early in your main() method using:
`Configuration.GLFW_LIBRARY_NAME.set("glfw_async");` per [the docs](https://github.com/LWJGL/lwjgl3/blob/master/doc/notes/3.3.1.md)

It's preferable not to be stuck in a situation where we're recommending XstartOnFirstThread. That approach has its own caveats, like being unable to use AWT/Swing at all.

**Comment by kyu-n** [comment]
> 

These glfw async libraries are fundamentally incompatible with the `-XstartOnFirstThread` JVM arg, which is why you were crashing in the first scenario.

> So I should simply add that line to my main method and that fixes it? Is this a "safe" solution, or more like a temporary hack?

I don't think I'd regard it as a temporary hack. It's a recent addition to lwjgl3 3.3.1 and I'd guess it was split to ensure that there's a fallback in case it's buggy. Not sure if lwjgl3 ever plans to make the async lib the default, though we haven't had any issues using it in testing.

**Comment by judos** [comment]
I experience also a freeze but already when application starts up. Here's a related issue on lwjgl issuetracker: https://github.com/LWJGL/lwjgl3/issues/619

The `org.lwjgl.system.Configuration.GLFW_LIBRARY_NAME.set("glfw_async");` crashes immediately when added to main with:
```
Process finished with exit code 133 (interrupted by signal 5: SIGTRAP)
```

**Comment by rotter** [comment]
Just adding some info to this issue:

I have just updated several things on my project (versions on build.gradles, gradle wrapper version, etc), and now that "hack" below causes my game not to launch, as soon as it reaches that line below the game closes down without any error messages. Since this line is before creating the Lwjgl3Application class, the game window doesnt even show up.

Configuration.GLFW_LIBRARY_NAME.set("glfw_async");

As soon as I commented out this line the game launched normally, and it seems the freeze I was getting originally when closing the game is gone also.

**Comment by tommyettinger** [comment]
Are you using glfw_async with -XstartOnFirstThread , or without? This whole thing is a total mess, and the approach I go with these days is just to include the [StartOnFirstThreadHelper](https://github.com/crykn/guacamole/blob/master/gdx-desktop/src/main/java/de/damios/guacamole/gdx/StartOnFirstThreadHelper.java) from the Guacamole helper library, which is just one file that can be copied into an LWJGL3 module.

**Comment by rotter** [comment]
Im not using any of those 2 commands anymore (glfw_async or XstartOnFirstThread) and it seems to be running fine. Im not sure if I just got lucky or if the updates on build.gradle fixed it.

What I did: I copied all the build.gradle wrapper androidmanifest etc from a new project created using the libgdx setup tool (and adjusted the project name/package name etc to my game). There were some other changes on the build.gradle structure as well (when compared to my project's build.gradle files), not only the libs versions.

**Comment by tommyettinger** [comment]
I've never heard of a Mac being able to run an LWJGL3 app/game out of the box without any handling for the first thread issue, which could be any of the glfw_async configuration, -XstartOnFirstThread argument (possibly via Guacamole), or gdx-lwjgl3-glfw-awt-macos dependency. Gradle also shouldn't be related, and should be unrelated entirely if you build a jar and launch that. I'm definitely curious what you have going that works with no extra changes to a gdx-setup project.

**Comment by rotter** [comment]
> I've never heard of a Mac being able to run an LWJGL3 app/game out of the box without any handling for the first thread issue, which could be any of the glfw_async configuration, -XstartOnFirstThread argument (possibly via Guacamole), or gdx-lwjgl3-glfw-awt-macos dependency. Gradle also shouldn't be related, and should be unrelated entirely if you build a jar and launch that. I'm definitely curious what you have going that works with no extra changes to a gdx-setup project.

You are correct, sorry. The -Xstart flag was present on the Gradle task. I tried to remove it now and it wouldnt launch ("Caused by: java.lang.IllegalStateException: GLFW may only be used on the main thread and that thread must be the first thread in the process. Please run the JVM with -XstartOnFirstThread. This check may be disabled with Configuration.GLFW_CHECK_THREAD0.").

**Comment by crykn** [comment]
Likely fixed by #7361.
